casper/src/main/scala/coop/rchain/casper/util/ProtoUtil.scala

```diff
@@ -422,4 +426,77 @@ object ProtoUtil {
     (missingParents union missingJustifications).toList
   }
 
+  // Return hashes of all blocks that are yet to be seen by the passed in block
+  def unseenBlockHashes[F[_]: Sync: BlockStore](
+      dag: BlockDagRepresentation[F],
+      block: BlockMessage
+  ): F[Set[BlockHash]] =
+    for {
+      latestMessages        <- dag.latestMessages
+      latestMessagesOfBlock <- ProtoUtil.toLatestMessage(block.justifications, dag)
+      result <- latestMessages.toList
```

Comment:
toList is inferior to toStream because it has to create an actual List object containing the data while the Stream is lazy and it only iterates through the underlying data structure